packages/taquito-michel-codec/src/michelson-types.ts

```diff
@@ -321,6 +324,7 @@ type DataList<T extends MichelsonData[]> = T & Node;
 export type MichelsonDataPair<T extends MichelsonData[]> = DataX<'Pair', T> | DataList<T>;
 export type MichelsonMapElt = PrimX<'Elt', [MichelsonData, MichelsonData]>;
 export type MichelsonMapEltList = List<MichelsonMapElt>;
+export type MichelsonLambdaRec = DataX<'Lambda_rec', [MichelsonLambdaRec]>
```

Comment:
wondering if this makes sense since it wouldnt be MichelsonData right? it would be itself recursively

Reply:
I don't think `MichelsonLambdaRec` should be inside `DataX`, but a list of Michelson instructions (there is probably an existing type for that).